core/src/main/java/io/questdb/ServerMain.java

```diff
@@ -336,6 +339,54 @@ public static void main(String[] args) throws Exception {
         }
     }
 
+    static void reportCrashFiles(CairoConfiguration configuration, Log log) {
+        final CharSequence dbRoot = configuration.getRoot();
+        final FilesFacade ff = configuration.getFilesFacade();
+        final int maxFiles = configuration.getMaxCrashFiles();
+
+        NativeLPSZ name = new NativeLPSZ();
+        try (
+                Path path = new Path().of(dbRoot).slash$();
+                Path other = new Path().of(dbRoot).slash$()
+        ) {
+            int plen = path.length();
+            AtomicInteger counter = new AtomicInteger(0);
+            FilesFacadeImpl.INSTANCE.iterateDir(
+                    path,
+                    (pUtf8NameZ, type) -> {
+                        if (Files.notDots(pUtf8NameZ)) {
+                            name.of(pUtf8NameZ);
+                            if (Chars.startsWith(name, "hs_err_pid")) {
+
+                                path.trimTo(plen).concat(pUtf8NameZ).$();
+
+                                boolean shouldRename = false;
+                                do {
+                                    other.trimTo(plen).concat("crash_").put(counter.getAndIncrement()).put(".log").$();
+                                    if (!ff.exists(other)) {
+                                        shouldRename = true;
+                                        break;
+                                    }
+                                } while (counter.get() < maxFiles);
+
+                                if (shouldRename && ff.rename(path, other) == 0 && counter.get() <= maxFiles) {
```

Comment:
minor: no point checking `counter.get() <= maxFiles` after rename already done
```suggestion
                                if (shouldRename && ff.rename(path, other) == 0) {
```

Reply:
good spot, fixed here: https://github.com/questdb/questdb/pull/2346/commits/a5b672cd99c9ec5aeb129b9ca7f38de0fbfe9461